### ISSUE TITLE
Don't know how to check if variable (when element class is org.elixir_lang.psi.impl.ElixirNoParenthesesManyStrictNoParenthesesExpressionImpl)

### ISSUE THREAD

**Issue by josuehenrique** [KronicDeth/intellij-elixir#422]
# Version

4.3.0

# Exception

## Message

Don't know how to check if variable (when element class is org.elixir_lang.psi.impl.ElixirNoParenthesesManyStrictNoParenthesesExpressionImpl)

### Excerpt

```
inner_join notifications, asspc
```
 Line(s) 8-8

## Stacktrace

```
org.elixir_lang.errorreport.Logger.error(Logger.java:48)
org.elixir_lang.errorreport.Logger.error(Logger.java:31)
org.elixir_lang.reference.Callable.error(Callable.java:471)
org.elixir_lang.reference.Callable.isVariable(Callable.java:345)
org.elixir_lang.reference.Callable.isVariable(Callable.java:573)
org.elixir_lang.reference.Callable.isVariable(Callable.java:339)
org.elixir_lang.reference.Callable.isVariable(Callable.java:558)
org.elixir_lang.reference.Callable.isVariable(Callable.java:339)
org.elixir_lang.psi.impl.ElixirPsiImplUtil.getUseScope(ElixirPsiImplUtil.java:2430)
org.elixir_lang.psi.impl.ElixirMatchedUnqualifiedNoArgumentsCallImpl.getUseScope(ElixirMatchedUnqualifiedNoArgumentsCallImpl.java:95)
com.intellij.psi.impl.search.PsiSearchHelperImpl.getUseScope(PsiSearchHelperImpl.java:79)
com.intellij.psi.search.searches.ReferencesSearch$SearchParameters.getEffectiveSearchScope(ReferencesSearch.java:107)
com.intellij.psi.impl.search.CachesBasedRefSearcher.processQuery(CachesBasedRefSearcher.java:55)
com.intellij.psi.impl.search.CachesBasedRefSearcher.processQuery(CachesBasedRefSearcher.java:21)
com.intellij.openapi.application.QueryExecutorBase.execute(QueryExecutorBase.java:95)
com.intellij.util.ExecutorsQuery.processResults(ExecutorsQuery.java:45)
com.intellij.util.AbstractQuery.forEach(AbstractQuery.java:79)
com.intellij.util.MergeQuery.processSubQuery(MergeQuery.java:85)
com.intellij.util.MergeQuery.forEach(MergeQuery.java:57)
com.intellij.util.UniqueResultsQuery.process(UniqueResultsQuery.java:66)
com.intellij.util.UniqueResultsQuery.forEach(UniqueResultsQuery.java:56)
com.intellij.util.UniqueResultsQuery.findAll(UniqueResultsQuery.java:79)
com.intellij.find.findUsages.FindUsagesHandler.findReferencesToHighlight(FindUsagesHandler.java:176)
com.intellij.codeInsight.daemon.impl.IdentifierHighlighterPass.a(IdentifierHighlighterPass.java:162)
com.intellij.codeInsight.daemon.impl.IdentifierHighlighterPass.getHighlightUsages(IdentifierHighlighterPass.java:139)
com.intellij.codeInsight.daemon.impl.IdentifierHighlighterPass.a(IdentifierHighlighterPass.java:195)
com.intellij.codeInsight.daemon.impl.IdentifierHighlighterPass.doCollectInformation(IdentifierHighlighterPass.java:106)
com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:70)
com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.b(PassExecutorService.java:434)
com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1034)
com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.c(PassExecutorService.java:427)
com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:494)
com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:443)
com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.a(PassExecutorService.java:426)
com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:402)
com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:154)
java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

**Comment by KronicDeth** [comment]
The `Strict` in `ElixirNoParenthesesManyStrictNoParenthesesExpressionImpl` indicates the element of part of the grammar where the native parser would error, but the IDE keeps parsing.  These elements are usually activated when you're typing or paste in a chunk of code that isn't natively valid yet.  Could you please try to reproduce the error by typing or pasting a little bit of code at a time.  Once you figure out how to trigger it, please post those steps.

**Comment by josuehenrique** [comment]
```
defmodule InfraTg.Query.AssociateUserNotification.Crud do
  import Ecto.Query
  alias InfraTg.Schema
  alias Schema.AssociateUserNotification
  alias Schema.Notification

  def list_messages do
    from associate_user_notification in AssociateUserNotification,
    inner_join notifications, assoc   # <- **_wrong syntax raises this error_**
  end
end
```

**Comment by KronicDeth** [comment]
Confirmed I can reproduce by pasting [sample code](https://github.com/KronicDeth/intellij-elixir/issues/422#issuecomment-246008280) and moving cursor to `assoc`.

**Comment by KronicDeth** [comment]
Fixed by #435
